src/main/java/fr/euphyllia/energie/utils/TeleportUtils.java

```diff
@@ -0,0 +1,21 @@
+package fr.euphyllia.energie.utils;
+
+import fr.euphyllia.energie.Energie;
+import io.papermc.paper.entity.TeleportFlag;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.player.PlayerTeleportEvent;
+
+import java.util.concurrent.CompletableFuture;
+
+public class TeleportUtils {
+    public static CompletableFuture<Boolean> teleportAsync(Entity entity, Location location) {
+        return Energie.isFolia() ? entity.teleportAsync(location) : CompletableFuture.completedFuture(entity.teleport(location));
+    }
+    public static CompletableFuture<Boolean> teleportAsync(Entity entity, Location location, PlayerTeleportEvent.TeleportCause cause) {
+        return Energie.isFolia() ? entity.teleportAsync(location, cause) : CompletableFuture.completedFuture(entity.teleport(location, cause));
+    }
+    public static CompletableFuture<Boolean> teleportAsync(Entity entity, Location location, PlayerTeleportEvent.TeleportCause cause, TeleportFlag flag) {
+        return Energie.isFolia() ? entity.teleportAsync(location, cause) : CompletableFuture.completedFuture(entity.teleport(location, cause, flag));
+    }
+}
```

Comment:
Add a possibility to check if it is Paper and the version used like here: https://forums.papermc.io/threads/paper-velocity-1-20-2.920/

So Energie.isPaper() and Energie.getMinecraftVersion() is greater than and equal to 1.13.2

Reply:
It might be better to use reflection to get the teleportAsync() method of the entity class?

Reply:
Yes, it can work, and if the class doesn't exist, then we use the basic teleport